### DISCUSSION TITLE
How to search files with a filename *searchstring* and a size larger than 100 KB (globally)

### DISCUSSION THREAD

**Discussion by rwiesbach** [community/community#137220] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have been trying to find out how to do it for over 1 hour now.
I assume that Code search would be the right tool, even though the file may not necessarily be code (but raw/blob, plaintext or something else)
https://github.com/search/advanced does not have a filesize, only a repo size.

Google results often date back many years and are related to an old search syntax:
filename: and size: are unknown quantifiers to Code Search, but mentioned in the google results "all over the place".
Even the Bing copilot suggests `filename:BLABLA size:>100KB`

**Comment by Yackens** [comment]
It sounds like you're trying to search for files based on their size within a GitHub repository, but unfortunately, GitHub's advanced search doesn't offer a direct way to filter files by size.

My advice would be that you clone the repository locally and use command-line tools to search for files by size:
```
git clone <repo_url>
find . -type f -size +1M   # Find files larger than 1MB
```

If you found this answer useful, please, mark my answer as the correct one to help me obtain GitHub achievement: Galaxy Brain 😄

**Reply by rwiesbach** [reply]
> It sounds like you're trying to search for files based on their size within a GitHub repository, but unfortunately, GitHub's advanced search doesn't offer a direct way to filter files by size.

well, not for a single repo but for all of (public) GitHub, so I cannot just clone it locally :)

**Comment by look** [comment, marked as answer]
You're right, the search syntax changed when we introduced the new code search. `filename` was replaced with the more general-purpose [`path` qualifier](https://docs.github.com/en/search-github/github-code-search/understanding-github-code-search-syntax#path-qualifier). The size filter is not supported at this time.

However, you can use the old syntax (including the `size:` qualifier) with the GitHub API. That may be a workaround for your use case.

**Reply by rwiesbach** [reply]
Would you kindly tell me how this example would like like when using the API, like which REST-endpoint is to be used? 
I guess it is easy if you are used to the API, but it would surely save quite some of my time as I do not intend to use the API for anything more than this. (I hope it can be used without generating API-Key and other auth-stuff involved?!)
Thank you.
